x-pack/plugins/ml/public/application/datavisualizer/index_based/page.tsx

```diff
@@ -165,40 +169,62 @@ export const Page: FC = () => {
     queryLanguage: initQueryLanguage,
   } = extractSearchData(currentSavedSearch);
 
-  const [searchString, setSearchString] = useState(initSearchString);
-  const [searchQuery, setSearchQuery] = useState(initSearchQuery);
-  const [searchQueryLanguage, setSearchQueryLanguage] = useState<SearchQueryLanguage>(
-    initQueryLanguage
-  );
-  const [samplerShardSize, setSamplerShardSize] = useState(defaults.samplerShardSize);
+  const searchQueryLanguage = dataVisualizerListState.searchQueryLanguage ?? initQueryLanguage;
+  const searchString = dataVisualizerListState.searchString ?? initSearchString;
+  const searchQuery = dataVisualizerListState.searchQuery ?? initSearchQuery;
+
+  const setSearchParams = (searchParams: {
+    searchQuery: Query['query'];
+    searchString: Query['query'];
+    queryLanguage: SearchQueryLanguage;
+  }) => {
+    setDataVisualizerListState({
+      ...dataVisualizerListState,
+      searchQuery: searchParams.searchQuery,
+      searchString: searchParams.searchString,
+      searchQueryLanguage: searchParams.queryLanguage,
+    });
+  };
+
+  const samplerShardSize =
+    dataVisualizerListState.samplerShardSize ?? restorableDefaults.samplerShardSize;
+  const setSamplerShardSize = (value: number) => {
+    setDataVisualizerListState({ ...dataVisualizerListState, samplerShardSize: value });
+  };
+
+  const visibleFieldTypes =
+    dataVisualizerListState.visibleFieldTypes ?? restorableDefaults.visibleFieldTypes;
+  const setVisibleFieldTypes = (values: string[]) => {
+    setDataVisualizerListState({ ...dataVisualizerListState, visibleFieldTypes: values });
+  };
+
+  const visibleFieldNames =
+    dataVisualizerListState.visibleFieldNames ?? restorableDefaults.visibleFieldNames;
+  const setVisibleFieldNames = (values: string[]) => {
+    setDataVisualizerListState({ ...dataVisualizerListState, visibleFieldNames: values });
+  };
+
+  const showEmptyFields =
+    dataVisualizerListState.showEmptyFields ?? restorableDefaults.showEmptyFields;
+  const toggleShowEmptyFields = () => {
+    setDataVisualizerListState({
+      ...dataVisualizerListState,
+      showEmptyFields: !dataVisualizerListState.showEmptyFields,
+    });
+  };
 
   // TODO - type overallStats and stats
```

Comment:
Can this `TODO` be removed - looks like you have typed `overallStats` at least in this PR.

Reply:
Removed here https://github.com/elastic/kibana/pull/85726/commits/13fa2c8f86d766e6db81a72638ed3dd8f968cedb